### PR TITLE
Create archive from cached paths

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [ ] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [ ] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)

### Changes

- Call `tar` and `zstd` with the required parameters to create a compressed archive from the input paths
- Expand relative paths (and `~/` references) before passing on the paths to compression
- Add E2E tests
- Update `go-steputils` with the latest changes around template parsing
- Add a new dependency for human-friendly file size printing

### Decisions

The tool `zstd` is added as a step `dep`, this will get preinstalled on the VMs soon.

The E2E tests are not verifying any of the outputs, just run the step. Once we have both a fully working save and restore step, we can have true end-to-end tests.

### CODE REVIEW

File: step.yml [hunk 1]
```diff
@@ -14,6 +14,12 @@ toolkit:
   go:
     package_name: github.com/bitrise-steplib/bitrise-step-save-cache
 
+deps:
+  brew:
+  - name: zstd
+  apt_get:
+  - name: zstd
```
Comment (hisaac) [inline step.yml:21]: My instinct would be to use a built-in compression binary for compressing files. I don't know much about `zstd` though, could you explain why you chose to use this over something built-in?

Comment (ofalvai) [reply step.yml:21]: I made some benchmarks of the different compression tools and options, see here: https://bitrise.atlassian.net/wiki/spaces/ENGI/pages/2351169600/RFC+New+key-based+caching+steps#Compression-benchmarks

The speedup is so significant that it's basically a free win

File: step/step.go [hunk 2]
```diff
@@ -47,42 +55,103 @@ func (step SaveCacheStep) ProcessConfig() (*Config, error) {
...
 	pathSlice := strings.Split(input.Paths, "\n")
 	for _, path := range pathSlice {
-		if exists, _ := step.pathChecker.IsPathExists(path); !exists {
+		absPath, err := step.pathModifier.AbsPath(path)
+		if err != nil {
+			step.logger.Warnf("Failed to parse path %s, error: %s", path, err)
+			continue
+		}
+
+		if exists, _ := step.pathChecker.IsPathExists(absPath); !exists {
```
[the remark this answers is not in the document]
Comment (ofalvai) [reply step/step.go]: Good catch, please take a look now, I made some adjustments. I intentionally don't want to abort the entire function from inside this for loop, I think the step should continue running despite these path errors, but we should log them as warnings.

[the remark this answers is not in the document]
Comment (hisaac) [reply step/step.go]: LGTM

File: step.yml (hunk 1 above)
Comment (ofalvai) [reply step.yml:21]: Yeah, I agree with all of this! Until then, nightly E2E tests will also catch a potential error like that.

**Pull request merged**